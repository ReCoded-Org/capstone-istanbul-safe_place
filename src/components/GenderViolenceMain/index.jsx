import React from "react";
import { Container, Row, Col } from "react-bootstrap";
import { Link } from "react-router-dom";
import { useTranslation } from "react-i18next";
import stopGenderViolenceImg from "../../images/home/stop-gender-violence.svg";
import "./index.scss";

const GenderViolenceMain = () => {
  const { t } = useTranslation();

  return (
    <Container fluid="md" className="main-container">
      <Row>
        <Col
          className="main-text"
          md={{ span: 6, order: "first" }}
          xs={{ span: 12, order: "last" }}
        >
          <div>
            <h1>
              {t("home.genderViolenceMain.headerPartOne")}
              <span>{t("home.genderViolenceMain.headerPartTwo")}</span>
            </h1>
          </div>
          <p>{t("home.genderViolenceMain.subHeader")}</p>
          <Link to="/seekhelp">
            <button type="button">{t("home.genderViolenceMain.button")}</button>
          </Link>
        </Col>
        <Col md={{ span: 6, order: "last" }} xs={{ span: 12, order: "first" }}>
          <img src={stopGenderViolenceImg} alt="Stop Gender Violence" />
        </Col>
      </Row>
    </Container>
  );
};

export default GenderViolenceMain;
