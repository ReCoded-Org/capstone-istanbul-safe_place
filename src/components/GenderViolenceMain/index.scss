@import "../../style/variables";
@import "../../style/buttons";
@import "../../style/bootstrapImports";

.main-container {
  background-color: $lightColor;

  .main-text {
    align-self: center;
    font-family: $mainFontFamily;
    font-size: 1.2rem;
    color: $primaryColor;

    p {
      margin-top: 1.4rem;
    }

    span {
      font-family: $secondaryFontFamily;
      font-weight: 800;
      text-transform: capitalize;
      margin-left: 1rem;
    }

    button {
      @include primaryBtn;
      margin-top: 1.4rem;
    }

    h1 {
      font-size: 3.2rem;
      width: 90%;
    }
  }

  img {
    width: 100%;
  }
}

@include media-breakpoint-down(lg) {
  .main-container {
    .main-text {
      margin-top: 1rem;
    }
  }
}

@include media-breakpoint-down(sm) {
  .main-container {
    .main-text {
      button {
        width: 100%;
      }
    }
  }
}
