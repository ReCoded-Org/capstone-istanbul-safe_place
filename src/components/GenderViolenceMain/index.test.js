import React from "react";
import renderer from "react-test-renderer";
import GenderViolenceMain from "./index";

it("renders correctly and matches the snapshot", () => {
  const tree = renderer.create(<GenderViolenceMain />).toJSON();
  expect(tree).toMatchSnapshot();
});
