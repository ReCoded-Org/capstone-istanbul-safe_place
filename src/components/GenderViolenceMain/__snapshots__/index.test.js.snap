// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches the snapshot 1`] = `
<div
  className="main-container container-md"
>
  <div
    className="row"
  >
    <div
      className="main-text col-md-6 col-12 order-md-first order-last"
    >
      <h1>
        You are not alone
        <br />
        Safe Place is
        <span>
           HERE
        </span>
      </h1>
      <p>
        Safe Place provides access to immediate help and support for victims of violence. We offer a shelter, psychological help, lawyers and doctors.
      </p>
      <button
        type="button"
      >
        Find Help
      </button>
    </div>
    <div
      className="col-md-6 col-12 order-md-last order-first"
    >
      <img
        alt="stop Gender Violence"
        src="stop-gender-violence.svg"
      />
    </div>
  </div>
</div>
`;
